DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,188 @@
+/*
+ *
+* This is a part of CTPPS offline software.
+* Author:
+*   Fabrizio Ferro (ferro@ge.infn.it)
+*   Enrico Robutti (robutti@ge.infn.it)
+*   Fabio Ravera   (fabio.ravera@cern.ch)
+*
+*/
+
+#ifndef DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+#define DataFormats_CTPPSReco_CTPPSPixelLocalTrack_H
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelRecHit.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSPixelDetId.h"
+
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/Math/interface/Vector3D.h"
+#include "DataFormats/Math/interface/Vector.h"
+#include "DataFormats/Math/interface/Error.h"
+#include "DataFormats/Math/interface/Matrix.h"
+
+
+//----------------------------------------------------------------------------------------------------
+
+class CTPPSPixelFittedRecHit: public CTPPSPixelRecHit
+{
+public:
+CTPPSPixelFittedRecHit(const CTPPSPixelRecHit &hit, const math::GlobalPoint &space_point_on_det, LocalPoint residual, LocalPoint pull) :
+  CTPPSPixelRecHit(hit), space_point_on_det_(space_point_on_det), residual_(residual), pull_(pull), isUsedForFit_(false), isRealHit_(false) {}
+    
+CTPPSPixelFittedRecHit() : CTPPSPixelRecHit(), residual_(LocalPoint(0,0)), pull_(LocalPoint(0,0)), isUsedForFit_(false), isRealHit_(false) {}
+    
+  virtual ~CTPPSPixelFittedRecHit() {}
+    
+  inline const math::GlobalPoint & getGlobalCoordinates() const { return space_point_on_det_; }
+  inline float getXResidual() const { return residual_.x(); }
+  inline float getYResidual() const { return residual_.y(); }
+        
+  inline float getXPull() const { return pull_.x(); }
+  inline float getYPull() const { return pull_.y(); }
+        
+  inline float getXPullNormalization() const { return residual_.x() / pull_.x(); }
+  inline float getYPullNormalization() const { return residual_.y() / pull_.y(); }
+
+  inline void setIsUsedForFit(bool usedForFit) {
+    if(usedForFit) isRealHit_ = true; 
+    isUsedForFit_ = usedForFit; 
+  }
+  inline bool getIsUsedForFit() const { return isUsedForFit_; }
+
+  inline void setIsRealHit(bool realHit) { 
+    if(!realHit) isUsedForFit_ = false;
+    isRealHit_ = realHit; 
+  }
+  inline bool getIsRealHit() const { return isRealHit_; }
+    
+private:
+  math::GlobalPoint space_point_on_det_ ;  ///< mm
+  LocalPoint residual_;  ///< mm
+  LocalPoint pull_    ;  ///< normalised residual
+  bool isUsedForFit_;
+  bool isRealHit_;
+};
+
+
+class CTPPSPixelLocalTrack: public CTPPSPixelFittedRecHit
```

Comment:
what is the reason for this inheritance?
Does a track have any meaning for inherited interface like getIsRealHit ?
This was not in the original submission and I'm guessing is not really needed.
Please remove or explain.